File: App.Contracts/Responses/PlayerResponse.cs
using App.Domain.Entities;

namespace App.Contracts.Responses;

public sealed record PlayerResponse(
    Guid Id,
    Guid? RoomId,
    bool IsLeader,
    bool Readiness,
    string PlayerName,
    int Score,
    string AvatarUrl,
    IEnumerable<Card> Cards,
    bool Move,
    int Money,
    bool InGame);

File: Web.ClientApp/src/contracts/player-response.ts
import {Card} from "./added-card-response";

export interface IPlayerResponse{
    id: string;
    roomId: string;
    isLeader: boolean;
    readiness: boolean;
    playerName: string;
    score: number;
    avatarUrl: string;
    cards: Array<Card>;
    move: boolean;
    money: number;
    inGame: boolean;
}

export type Player = IPlayerResponse;